src/legacy/server/index_patterns/service/lib/field_capabilities/field_caps_response.js

```diff
@@ -120,7 +120,26 @@ export function readFieldCapsResponse(fieldCapsResponse) {
       aggregatable: isAggregatable,
       readFromDocValues: shouldReadFieldFromDocValues(isAggregatable, esType),
     };
-  }).filter(field => {
+  });
+
+  // Get all types of sub fields. These could be multi fields or children of nested/object types
+  const subFields = kibanaFormattedCaps.filter(field => {
+    return field.name.includes('.');
+  });
+
+  // Discern which sub fields are multi fields. If the parent field is not an object or nested field
+  // the child must be a multi field.
+  subFields.forEach(field => {
+    const parentFieldName = field.name.split('.').slice(0, -1).join('.');
+    const parentFieldCaps = kibanaFormattedCaps.find(caps => caps.name === parentFieldName);
+
+    if (!['object', 'nested'].includes(parentFieldCaps.type)) {
```

Comment:
Should you check that `parentFieldCaps` is not undefined since `find` could return undefined if parentFieldName can not be found?

Reply:
You should probably also check if the type is `'join'` as well: https://www.elastic.co/guide/en/elasticsearch/reference/current/parent-join.html

Reply:
Added check for undefined.

I wasn't sure how `join` would be reported in field_caps so I tested it out. Looks like we don't need to worry about it. It's a bit different because the relationship is between documents, not fields, so the relationship is not represented with a dot in the field name. Here's a sample response with the extraneous meta fields removed:

```
{
  "fields" : {
    "my_join_field#question" : {
      "parent" : {
        "type" : "parent",
        "searchable" : true,
        "aggregatable" : true
      }
    },
    "_parent_join" : {
      "parent_join" : {
        "type" : "parent_join",
        "searchable" : false,
        "aggregatable" : false
      }
    },
    "my_join_field" : {
      "join" : {
        "type" : "join",
        "searchable" : true,
        "aggregatable" : true
      }
    },
    "text" : {
      "text" : {
        "type" : "text",
        "searchable" : true,
        "aggregatable" : false
      }
    },
    "text.keyword" : {
      "keyword" : {
        "type" : "keyword",
        "searchable" : true,
        "aggregatable" : true
      }
    },
  }
}
```